truss/cli/resolvers/training_project_team_resolver.py

```diff
@@ -0,0 +1,71 @@
+"""Team resolution logic for training projects."""
+
+from typing import Optional
+
+import click
+
+from truss.cli import remote_cli
+from truss.remote.baseten.remote import BasetenRemote
+
+
+def resolve_training_project_team_name(
+    remote_provider: BasetenRemote,
+    provided_team_name: Optional[str],
+    existing_project_name: Optional[str] = None,
+    existing_teams: Optional[dict[str, dict[str, str]]] = None,
+) -> Optional[str]:
```

Comment:
nit: we could have this return a tuple of `team_name`, `team_id` to save us a lookup in `push_training_job`, i think its fine to have `inquire_team` return this tuple as well